.github/workflows/gitsecrets.yml

```diff
@@ -0,0 +1,21 @@
+name: GitSecretsScan
+
+on: [push, pull_request]
+
+jobs:
+  git-secret-check:
+    name: Git Secrets Scan
+    runs-on: macos-latest
```

Comment:
is there a git secrets check we can run on ubuntu? my understanding is macos instances are more costly and less available.

Reply:
To install git secret linux instance requires sudo access --- I am not sure if that's available in these workflow workers, I can try that

Reply:
Just did a test for that, the workflow is able to use sudo. I will update to use ubuntu.